app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailablePresenter.java

```diff
@@ -0,0 +1,49 @@
+package io.intrepid.contest.screens.conteststatus.resultsavailable;
+
+import android.support.annotation.NonNull;
+
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BasePresenter;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.rest.ContestResponse;
+import io.reactivex.functions.Consumer;
+import timber.log.Timber;
+
+public class ResultsAvailablePresenter extends BasePresenter<ResultsAvailableContract.View>
+        implements ResultsAvailableContract.Presenter {
+
+    public ResultsAvailablePresenter(@NonNull ResultsAvailableContract.View view,
+                                     @NonNull PresenterConfiguration configuration) {
+        super(view, configuration);
+    }
+
+    @Override
+    public void onViewCreated() {
+        super.onViewCreated();
+
+        view.requestContestDetails(onContestDetailsRetrieved(), onContestDetailsError());
+    }
+
+    @NonNull
+    private Consumer<ContestResponse> onContestDetailsRetrieved() {
+        return response -> view.showContestName(response.contest.getTitle());
+    }
+
+    @NonNull
+    private Consumer<Throwable> onContestDetailsError() {
```

Comment:
Why not have your presenter implement Consumer<ContestResponse> and Consumer<Throwable> rather than making those two callbacks a parameter of some of the presenter and view methods. I think  view.requestContestDetails(this, this); look a bit cleaner.